src/tools/dev/xml/GeneratePython.h

```diff
@@ -1903,7 +1903,7 @@ class AttsGeneratorAtt : public virtual Att , public virtual PythonGeneratorFiel
             c << "                            if(badInput)" << Endl;
             c << "                            {" << Endl;
             c << "                                delete [] C;" << Endl;
-            c << "                                return NULL;" << Endl;
+            c << "                                return PyExc_ValueError;" << Endl;
```

Comment:
`badInput` in the code above is actually being used for multiple cases of the new checks being proposed here. One is for incorrect length (which we propose to handle as `ValueError` here) and the other is for an incorrect type (`TypeError`). This should be corrected.

Reply:
I adjusted logic  to address this